ls1intum/Artemis
# [Feature] Show more exercise information in the ExerciseGroupComponent

### Checklist

### Motivation and Context
We want to display more information about the exercises in the exercise group view. This should include the diagram type, the repositories and buildplan URLs, similar to how it is displayed in the course management view

### Description
- [x] [Client] Extend the ExerciseGroupComponent HTML table to include more details about the exercises
- [x] [Client] Only show columns relevant to the existing exercises
- [x] [Client] Fixed a bug on text exercise creation where the assessment type is undefined if the corresponding checkbox hasn't been touched
- [x] [Server] Attach the template and solution participation to the programming exercises
- [x] [Server] Attach the quiz questions to the quiz exercises


### Steps for Testing
1. Log in to Artemis
2. Navigate to Course Administration -> Exam
3. Create exercises grouped liked seen in the screenshots
--> Verify that the information is displayed correctly
4. Delete exercises in the mixed group one-by-one
--> Verify that the table only shows columns relevant for the existing exercises

### Screenshots
![group_mixed](https://user-images.githubusercontent.com/16407766/90561872-a1dfbb80-e1a1-11ea-84a0-e2f2360396e0.PNG)
Mixed group

![group_mod](https://user-images.githubusercontent.com/16407766/90561899-ab692380-e1a1-11ea-8bf2-c247ee4ab258.PNG)
Only modeling

![group_programming](https://user-images.githubusercontent.com/16407766/90561913-b3c15e80-e1a1-11ea-8e0d-ed5409f7e8f0.PNG)
Only programming

![group_quiz](https://user-images.githubusercontent.com/16407766/90561940-bf148a00-e1a1-11ea-8f44-78298e933bc2.PNG)
Only quiz

![group_text](https://user-images.githubusercontent.com/16407766/90561955-c6d42e80-e1a1-11ea-87b1-b48be9be57d4.PNG)
Only text
